test/browser-only/constructLayoutEngine-browser.test.js

```diff
@@ -0,0 +1,278 @@
+    /** @type {import('../../src/constructRoutes').ResolvedRoutesConfig} */
+    const routes = {
+      containerEl: "body",
+      base: "/",
+      mode: "history",
+      routes: [{ type: "application", name: "@org-name/header" }],
+    };
+
+    layoutEngine = constructLayoutEngine({
+      routes,
+    });
+
+    layoutEngine.activate();
+    expect(layoutEngine.isActive()).toBe(true);
+    layoutEngine.activate();
+    expect(layoutEngine.isActive()).toBe(true);
+
+    layoutEngine.deactivate();
+    expect(layoutEngine.isActive()).toBe(false);
+    layoutEngine.deactivate();
+    expect(layoutEngine.isActive()).toBe(false);
+
+    layoutEngine.activate();
+    expect(layoutEngine.isActive()).toBe(true);
+  });
+
+  it(`can successfully construct a layout engine and respond to routing events`, () => {
+    /** @type {import('../../src/constructRoutes').ResolvedRoutesConfig} */
+    const routes = {
+      containerEl: "body",
+      base: "/",
+      mode: "history",
+      routes: [
+        { type: "application", name: "@org-name/header" },
+        {
+          type: "route",
+          path: "app1",
+          routes: [{ type: "application", name: "@org-name/app1" }],
+        },
+        { type: "application", name: "@org-name/footer" },
+      ],
+    };
+
+    layoutEngine = constructLayoutEngine({
+      routes,
+    });
+
+    // start at / route
+    let headerEl = document.getElementById(
+      `single-spa-application:@org-name/header`
+    );
+    let app1El = document.getElementById(
+      `single-spa-application:@org-name/app1`
+    );
+    let footerEl = document.getElementById(
+      `single-spa-application:@org-name/footer`
+    );
+
+    expect(headerEl).toBeInTheDocument();
+    expect(app1El).not.toBeInTheDocument();
+    expect(footerEl).toBeInTheDocument();
+
+    expect(headerEl.compareDocumentPosition(footerEl)).toBe(
+      Node.DOCUMENT_POSITION_FOLLOWING
+    );
+    expect(document.body).toMatchSnapshot();
+
+    // transition to /app1 route
+    history.pushState(history.state, document.title, "/app1");
+    window.dispatchEvent(
+      new CustomEvent("single-spa:before-mount-routing-event")
+    );
+
+    headerEl = document.getElementById(
+      `single-spa-application:@org-name/header`
+    );
+    app1El = document.getElementById(`single-spa-application:@org-name/app1`);
+    footerEl = document.getElementById(
+      `single-spa-application:@org-name/footer`
+    );
+
+    expect(headerEl).toBeInTheDocument();
+    expect(app1El).toBeInTheDocument();
+    expect(footerEl).toBeInTheDocument();
+
+    expect(headerEl.compareDocumentPosition(footerEl)).toBe(
+      Node.DOCUMENT_POSITION_FOLLOWING
+    );
+    expect(headerEl.compareDocumentPosition(app1El)).toBe(
+      Node.DOCUMENT_POSITION_FOLLOWING
+    );
+    expect(footerEl.compareDocumentPosition(app1El)).toBe(
+      Node.DOCUMENT_POSITION_PRECEDING
+    );
+    expect(document.body).toMatchSnapshot();
+
+    // transition back to / route
```

Comment:
I don't actually see a change to the `history` here.
I think we need to add `history.pushState(history.state, document.title, "/");` here

Reply:
👍 good catch, will update